ts/spi/provider_frame_test.ts

```diff
@@ -300,6 +300,40 @@ describe('ProviderFrame', () => {
                msg.retrieveMessage(requestId, passwordOnlyRequest));
          });
 
+      it('should cancel a credential request', async function(done) {
+        credentialDataProvider.credentials = [alicePwdCred];
+
+        (interactionProvider.showAutoSignIn as jasmine.Spy)
+            .and.callFake(
+                (credential: Credential,
+                 displayCallbacks: DisplayCallbacks) => {
+                  expect(credential).toBe(alicePwdCred);
+                  // return a promise that's not going to resolve
+                  return new Promise<void>((resolve, reject) => {});
+                });
+
+        let credentialPromise = new PromiseResolver<void>();
+        let cancellationPromise = new PromiseResolver<void>();
+
+        clientChannel.listen(msg.RPC_MESSAGE_TYPES.credential, (data) => {
+          expectMessageContents(
+              data,
+              msg.credentialResultMessage(
+                  requestId, undefined as any as Credential));
```

Comment:
😄 The `expectMessageContents(...)` method expects an instance of type `Credential`. This is the only way for it to understand that the result of a cancelled action will be an undefined.